File: swarmmanagercontroller/src/main/java/co/uk/swarmbit/docker/api/networks/NetworksApi.java
package co.uk.swarmbit.docker.api.networks;

import co.uk.swarmbit.docker.api.common.json.NetworkJson;
import co.uk.swarmbit.docker.api.common.json.NetworkCreateResponseJson;
import co.uk.swarmbit.docker.api.networks.parameters.NetworkCreateParameters;
import co.uk.swarmbit.docker.api.networks.parameters.NetworkInspectParameters;
import co.uk.swarmbit.docker.api.networks.parameters.NetworkListParameters;

import java.util.List;

public interface NetworksApi {

    List<NetworkJson> listNetworks(String swarmId, NetworkListParameters parameters);

    NetworkJson inspectNetwork(String swarmId, String id, NetworkInspectParameters parameters);

    NetworkCreateResponseJson createNetwork(String swarmId, NetworkCreateParameters parameters);

    void deleteNetwork(String swarmId, String id);

}

File: swarmmanagercontroller-new/src/main/kotlin/com/swarmbit/docker/api/networks/NetworksApi.kt
package com.swarmbit.docker.api.networks

import com.swarmbit.docker.api.networks.model.Network
import com.swarmbit.docker.api.networks.model.NetworkCreateResponse
import com.swarmbit.docker.api.networks.parameters.NetworksCreateParameters
import com.swarmbit.docker.api.networks.parameters.NetworksInspectParameters
import com.swarmbit.docker.api.networks.parameters.NetworksListParameters

interface NetworksApi {

    fun listNetworks(swarmId: String, parameters: NetworksListParameters): List<Network>

    fun inspectNetwork(swarmId: String, id: String, parameters: NetworksInspectParameters): Network

    fun createNetwork(swarmId: String, parameters: NetworksCreateParameters): NetworkCreateResponse

    fun deleteNetwork(swarmId: String, id: String)
}